testing/fpgad/test_evt_c.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright(c) 2018, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+extern "C" {
+
+#include <json-c/json.h>
+#include <uuid/uuid.h>
+
+#include "fpgad/log.h"
+#include "fpgad/srv.h"
+#include "fpgad/errtable.h"
+#include "fpgad/ap6.h"
+
+struct client_event_registry *register_event(int conn_socket, int fd,
+	                                     fpga_event_type e, const char *device);
+
+void unregister_all_events(void);
+
+void evt_notify_error(const struct fpga_err *e);
+
+void evt_notify_ap6_and_null(const struct fpga_err *e);
+
+}
+
+#include <config.h>
+#include <opae/fpga.h>
+
+#include <array>
+#include <cstdlib>
+#include <unistd.h>
+#include <sys/eventfd.h>
+#include <poll.h>
+#include "gtest/gtest.h"
+#include "test_system.h"
+
+using namespace opae::testing;
+
+class fpgad_evt_c_p : public ::testing::TestWithParam<std::string> {
+ protected:
+  fpgad_evt_c_p()
+      : tmpfpgad_log_("tmpfpgad-XXXXXX.log") {}
+
+  virtual void SetUp() override {
+    tmpfpgad_log_ = mkstemp(const_cast<char *>(tmpfpgad_log_.c_str()));
```

Comment:
This may get flagged as a memory leak. That's one of the reasons why I moved away from doing this.
A better approach would be something like:
```C++
  char tmpname[]{ "tmpfpgad-XXXXXX.log" };
  tmpfpgad_log_ = mkstemp(tmpname);
```

Reply:
I was just looking at mkstemp.  The logic we're using is wrong, because mkstemp returns an open fd to the file, not the substituted name.  Thinking of how to solve this.